File: Best Time to Buy and Sell Stock II.java
class Solution {
    public int maxProfit(int[] prices) {
        int n = prices.length, lastBuy = -prices[0], lastSold = 0;
        if(n == 0) return 0;
        
        for(int i = 1; i < n; i++){
            int curBuy = Math.max(lastBuy, lastSold - prices[i]);
            int curSold = Math.max(lastSold, lastBuy + prices[i]);
            lastBuy = curBuy;
            lastSold = curSold;
        }
        return lastSold;
    }
}

File: Best Time to Buy and Sell Stock II.js
/**
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function(prices) {
    if(prices.length < 1) return 0;
    
    let maxprofit = 0;
    for(let i = 1; i < prices.length; i++){
        maxprofit += Math.max(0, prices[i] - prices[i-1])
    }
    return maxprofit;
};
